packages/devtools/lib/src/timeline/event_details.dart

```diff
@@ -187,54 +180,127 @@ class EventDetails extends CoreElement {
 }
 
 class _UiEventDetails extends CoreElement {
-  _UiEventDetails(TimelineController timelineController) : super('div') {
+  _UiEventDetails(this._timelineController) : super('div') {
     layoutVertical();
     flex();
 
     add([
-      flameChart = CpuFlameChart(timelineController),
+      flameChart = CpuFlameChart(_timelineController),
       bottomUp = CpuBottomUp()..attribute('hidden', true),
-      callTree = CpuCallTree()..attribute('hidden', true),
+      callTree = CpuCallTree(_timelineController)..attribute('hidden', true),
     ]);
   }
 
+  final TimelineController _timelineController;
+
   CpuFlameChart flameChart;
 
   CpuBottomUp bottomUp;
 
   CpuCallTree callTree;
 
-  Future<void> showTab(EventDetailsTabType tabType) async {
+  bool showingMessage = false;
+
+  void showTab(EventDetailsTabType tabType) {
+    // If we are showing a message, we do not want to show any other views.
+    if (showingMessage) return;
+
     switch (tabType) {
       case EventDetailsTabType.flameChart:
```

Comment:
this logic can be simplified.
consider changing show() and hide() . to a visibility header.
Then this whole switch statement could just be
List tabs = [flameChart, bottomUp, callTree];
for (tab in tabs) {
  tab.visible = tabType == tab.tabType;
}
You could also keep the show and hide methods and get most of the benefit.

Reply:
kept show and hide methods, but put them in a base class `CpuProfilerView`. Simplified logic here as you suggested.

Reply:
ok that is fine. disregard  the other comment about show and hide.

Reply:
logic still seems like it has O(N^2) calls to show and hide :)
you can still make it programatic even without adding a `show` setter.

Reply:
I need a show() method because each view does more than just `hidden(false]` on show. The `show` method is now in the base class to share the logic. I don't think it is O(N^2) because we are calling either show or hide only once for each of the three views ==> O(N).